poet/poet.py

```diff
@@ -94,22 +94,35 @@ class PackageMetadata:
     homepage: str
     url: str
     checksum: str
-    checksum_type: str
+    checksum_type: str = "sha256"
 
-def get_download_url_from_pip_source_file(module: str, pip_source_file: Path, output_dir=os.getenv("PIP_SOURCE_DIR")) -> str:
+    def __getitem__(self, idx):
+        try:
+            return getattr(self, idx)
+        except AttributeError as err:
+            raise KeyError(idx) from err
+
+    def __setitem__(self, idx, val):
+        setattr(self, idx, val)
+
+
+
+def get_download_url_from_pip_source_file(module: str, pip_source_file: Path, output_dir: Path) -> str:
     """
     Returns the download URL for the pip source distribution.
     This method will download the pip package from the source distribution. 
 
     The standard out of this command contains an obfuscated URL and a regular URL that points to a .tar.gz file.
 
-    Args:
+    Arguments:
         module (str): The name of the module to download.
         output_dir (str): The directory to download the module to.
 
     Returns:
         str: The download URL for the pip source distribution.
     """
+
+    output_dir = output_dir if output_dir != None else Path(os.getenv("PIP_SOURCE_DIR"))
```

Comment:
the joy of using a type system: you don't have to do this sort of thing any more!!!!!

your method signature says that `output_dir` is a `Path`.  not an `Optional[Path]`, but a `Path`.  that means that by definition it will never be `None`, b/c `None` is not a `Path`.

so instead you could do something even more useful, like
```suggestion
    if not output_dir.is_dir():
        output_dir = Path(os.getenv("PIP_SOURCE_DIR"))
```